### PR TITLE
Prepare v0.20.0

### PR DESCRIPTION
**Pull request by lukpueh**
*__NOTE:__ This will be the final release of python-tuf that includes the legacy implementation code. Please see the [*1.0.0 announcement*](1.0.0-ANNOUNCEMENT.md) page for more details about the next release and the deprecation of the legacy implementation, including migration instructions.*

### Added
* metadata API: misc input validation (#1630, #1688, #1668, #1672, #1690)
* doc: repository library design document and ADR (#1693)
* doc: 1.0.0 announcement (#1706)
* doc: misc docstrings in metadata API (#1620)
* doc: repository and client examples (#1675, #1685, #1700)
* test: ngclient key rotation (#1635, #1649, #1691)
* test: ngclient top-level role update (#1636)
* test: ngclient non-consistent snapshot (#1666, #1705)
* test: more lint/type checks and auto-formatting (#1658, #1664, #1659, #1674,
        #1677, #1687, #1699, #1701, #1708, #1710, #1720)
* build: Python 3.10 support (#1628)

### Changed
* ngclient: misc API changes (#1604)
* ngclient: avoid re-loading verified targets metadata (#1593)
* ngclient: implicitly call refresh() (#1654)
* ngclient: return loaded metadata (#1680)
* ngclient: skip visited nodes on delegation tree traversal (#1683)
* ngclient: remove URL normalisation (#1686)
* build: modernise packaging configuration (#1626)
* build: bump dependencies (#1609, #1611, #1616, #1621)
* build: limit GitHub Action token visibility and permissions (#1652, #1663)
* test: misc test changes (#1715, #1670, #1671, #1631, #1695)

### Removed
* doc: obsolete roadmap (#1698)

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+
+## v0.20.0
+
+*__NOTE:__ This will be the final release of python-tuf that includes the
+legacy implementation code. Please see the [*1.0.0
+announcement*](1.0.0-ANNOUNCEMENT.md) page for more details about the next
```
Comment (jku) [inline docs/CHANGELOG.md:8]: Not sure if that was a mistake but the 1.0.0 announcement ended up in root dir so this link does not work. Do we move the announcement?

**Review by jku — COMMENTED** [review]
* #1722 looks like a blocker
* announcement link is wrong (see below)

otherwise LGTM

File: docs/CHANGELOG.md (hunk 1 above)
Comment (lukpueh) [reply docs/CHANGELOG.md:8]: 🤦  Thanks for catching this!

**Comment by jku** [comment]
I've got an updated branch in https://github.com/jku/python-tuf/tree/release-0.20.0 but 
* I'd like to stop saying ngclient is not ready for prod: I've updated README in the branch
* but if I do that, then I don't want to break ngclient API even in a minor way after that unless I really have to...
* which means we should maybe handle the argument name change before 0.20: https://github.com/theupdateframework/python-tuf/pull/1731?

**Comment by lukpueh** [comment]
> I've got an updated branch in https://github.com/jku/python-tuf/tree/release-0.20.0 but
> * I'd like to stop saying ngclient is not ready for prod: I've updated README in the branch

Cool, thanks! Do you want to merge the README change with this PR, or create a separate one? Also, while changing the README, would you mind adding the cncf python-tuf slack channel to the contact section? I'm pointing migration help seekers to that section in the 1.0.0-ANNOUNCEMENT doc, and I think the slack channel would be a good way to reach out to us.

> * but if I do that, then I don't want to break ngclient API even in a minor way after that unless I really have to...

I am not so concerned about minor breaking changes prior to 1.0.0. But yeah, ...

> * which means we should maybe handle the argument name change before 0.20: [ngclient: Rename constructor arg #1731](https://github.com/theupdateframework/python-tuf/pull/1731)?

... the rename can easily land with 0.20 (just merged).

**Comment by jku** [comment]
I can make a new PR, thanks

**Comment by jku** [comment]
I'll close this as superseded by #1732 (but feel free to redo if needed @lukpueh i'm just hopefully helping out)

**Pull request closed without merging**